### PR TITLE
Predefined quantizer

### PR DESCRIPTION
**Pull request by jerch**
PR to make the SIXEL creation less painful and without deep knowledge about image processing:

- include quantizer in lib
- limited dithering suitable for most pictures
- convenient function `img2sixel` to use the internal quantizer
- major performance improvement (up to 10x faster)

### CODE REVIEW

File: upng.js [hunk 1]
```diff
@@ -0,0 +1,1046 @@
+
```
Comment (Richienb) [inline upng.js:1]: Can't you just add a dependency for [upng-js](https://www.npmjs.com/package/upng-js) instead?

Comment (jerch) [reply upng.js:1]: Nope, it has no properly exported package anywhere, just clones from random persons :(

Thus it is safer to rely on commit export from repo.

**Comment by jerch** [comment]
@Richienb I think I gonna remove UPNG as quantizer again and will rely on octree quantization for the internal quantizer only. The result will be slightly worse in terms of optimized palette colors, but it runs like >5 times faster. It also tends to perserve more colors from the colorspace, thus will show still acceptible result with smaller palettes.

If someone wants better adaptive algos - the lib will maintain the raw pixel level interface to hook in any other/better quantizer.

**Comment by Richienb** [comment]
@jerch Can we get an actual number on the performance differences?

**Comment by jerch** [comment]
@Richienb Not yet, the impl still needs cleanup/opts for that.

**Comment by Richienb** [comment]
@jerch I think we can merge this pr and publish it to npm.

**Comment by jerch** [comment]
@Richienb Yes. I have currently no time to investigate into a more self contained version of the quantizer, thus I gonna cleanup things and push it as it is (alpha).

Missing bits:
- transparent pixels are not correctly handled atm by the indexed encoder
- dithering may introduce pixel artefacts at the beginning (not an issue with photos though, but annoying with polished graphics)

**Pull request merged**